src/Components/Admins/Form/index.jsx

```diff
@@ -1,21 +1,57 @@
 import { useState, useEffect } from 'react';
+import { Link, useParams, useHistory } from 'react-router-dom';
+import { Input } from 'Components/Shared/Inputs';
 import { useDispatch, useSelector } from 'react-redux';
-import { getAdmins, updateAdmin, addAdmin } from '../../../redux/admins/thunks';
+import { getAdmins, updateAdmin, addAdmin } from 'redux/admins/thunks';
+import { useForm } from 'react-hook-form';
+import { joiResolver } from '@hookform/resolvers/joi';
+import Joi from 'joi';
 import styles from './adminsForm.module.css';
-import { Input } from '../../Shared/Inputs';
-import { Link, useParams, useHistory } from 'react-router-dom';
-import Button from '../../Shared/Button';
-import Modal from '../../Shared/Modal';
-import * as actionsConstants from '../../../redux/admins/actions';
+import Button from 'Components/Shared/Button';
+import Modal from 'Components/Shared/Modal';
+import * as actionsConstants from 'redux/admins/actions';
 
 const Form = () => {
+  const history = useHistory();
+  const { id } = useParams();
+  const dispatch = useDispatch();
   const { isPending, data, error, errorSwitch, message } = useSelector((state) => state.admins);
   const [inputs, setInputs] = useState({});
   const [successModal, setSuccesModal] = useState(false);
   const [errorModal, setErrorModal] = useState(false);
-  const history = useHistory();
-  const dispatch = useDispatch();
-  const { id } = useParams();
+
+  const RGXPassword = /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/;
+  const RGXEmail = /^[^@]+@[^@]+\.[a-zA-Z]{2,}$/;
+  const schema = Joi.object({
+    firstName: Joi.string().min(3).max(15),
+    lastName: Joi.string().min(3).max(15),
+    dni: Joi.string().length(9),
+    phone: Joi.string().length(10),
+    email: Joi.string().regex(RGXEmail).required().messages({
+      'string.pattern.base': 'Email must be in a valid format'
+    }),
+    city: Joi.string(),
+    password: Joi.string().regex(RGXPassword).min(8).required().messages({
+      'string.pattern.base':
+        'Password must contain at least one uppercase letter, one lowercase letter, and one digit'
```

Comment:
Check this password message validation, please..
Include that it is required at least 8 characters. 
Instead of '**_one digit_**', I think you mean '**_one special character_**'.